src/Data/Array.purs

```diff
@@ -1192,6 +1195,12 @@ unzip xs =
     snds' <- STA.unsafeFreeze snds
     pure $ Tuple fsts' snds'
 
+-- | Returns true if at least one array element satisfy the given predicate.
```

Comment:
We want “satisfies” here rather than “satisfy” for subject-verb agreement. Also, I think it would be nice to explain why we have specialised HeytingAlgebra to Boolean?

Reply:
A couple of simple examples would also be nice, I guess

Reply:
I updated the documentation, added some examples I fancied more than the tests and updated the tests.